sdk/js/src/User.ts

```diff
@@ -26,13 +26,13 @@ export class DVCPopulatedUser implements DVCUser {
     readonly sdkType: 'client' | 'server'
     readonly sdkVersion: string
 
-    constructor(user: DVCUser, options: DVCOptions, staticData?: StaticData) {
+    constructor(user: DVCUser, options: DVCOptions, staticData?: StaticData, anonymousUserId?: string) {
         if (!user.user_id && !user.isAnonymous) {
             throw new Error('Must have a user_id, or have "isAnonymous" set on the user')
         }
 
-        this.user_id = !user.user_id ? uuidv4() : user.user_id
-        this.isAnonymous = user.user_id ? false : user.isAnonymous
+        this.user_id =  user.isAnonymous ? anonymousUserId || uuidv4() : user.user_id || uuidv4()
```

Comment:
If the user is not anonymous they always have an id
```suggestion
        this.user_id =  user.isAnonymous ? anonymousUserId || uuidv4() : user.user_id
```

Reply:
yes, but there's a typescript error, `user_id` type is `undefined | string` so it will complain if not added a default value. I took a reference from android sdk and it uses `uuidv4()` to be the default value

Reply:
oh nvm then, you can leave it as is then